big-practice/src/js/pages/dashboard.page.js

```diff
@@ -1,114 +1,75 @@
+import { apiService } from '../service/api.service';
+import { DocumentHelper } from '../helpers/document.helper';
+import { ModalHelper } from '../helpers/modal.helper';
+import { MOVIES_API } from '../constants/url-api.constant';
+import { EMPTY_TEXT, MESSAGES } from '../constants/message.constant';
+import { USERNAME_KEY, LOGIN_PAGE } from '../constants/app.constant';
+
+class Dashboard {
+  // Query elements
+  tableBody = document.getElementById('tableBody');
+  nameMovieInput = document.getElementById('nameMovie');
+  directorInput = document.getElementById('director');
+  nationInput = document.getElementById('nation');
+  accountName = document.querySelector('.account-name');
+  form = document.querySelector('.form');
+  modalForm = document.querySelector('.modal-form');
+  modalWarning = document.querySelector('.modal-warning');
+  layoutMainAddBtn = document.querySelector('.layout-main .add-btn');
+  modalFormCreateBtn = document.querySelector('.modal-form .create-btn');
+  modalFormUpdateBtn = document.querySelector('.modal-form .update-btn');
+  modalFormCancelBtn = document.querySelector('.modal-form .cancel-btn');
+  modalWarningCancelBtn = document.querySelector('.modal-warning .cancel-btn');
+  modalWarningDeleteBtn = document.querySelector('.modal-warning .delete-btn');
+  logoutBtn = document.querySelector('.site-header .logout-btn');
+
+  constructor() {
+    this.checkUserLogin();
+    this.showUserName();
+    this.renderTable();
+    this.handleUserLogout();
+  }
 
-/**
- * Takes data from the API and displays it on a table in HTML
- */
-const renderTable = () => {
-  fetch(MOVIES_API)
-    // Parses JSON response into native JavaScript objects
-    .then((response) => response.json())
-    .then((movies) => {
+  // /**
+  //  * Modal form update will appear when clicking the update button in the table
+  //  *
+  //  * @param {element} item - Table update button
+  //  */
+  // tableUpdateBtn(item) {
+  //   const movieId = item.dataset.id;
+
+  //   showModal(modalForm);
+  //   getApi(`${MOVIES_API}/${movieId}`, (movieData) => {
+  //     nameMovieInput.value = movieData.name;
+  //     directorInput.value = movieData.director;
+  //     nationInput.value = movieData.nation;
+  //     form.setAttribute('data-id', movieId);
+  //   });
+  // }
+
+  // /**
+  //  * Modal warning will appear when clicking delete button in the table
+  //  *
+  //  * @param {element} item - Table delete button
+  //  */
+  // tableDeleteBtn(item) {
+  //   const movieId = item.dataset.id;
+
+  //   showModal(modalWarning);
+  //   modalWarningDeleteBtn.setAttribute('data-id', movieId);
+  // }
+
+  /**
+   * Takes data from the API and displays it on a table in HTML
+   */
+  async renderTable() {
+    try {
+      const result = await apiService.get(MOVIES_API);
       let tableTemplate = '';
 
-      movies.forEach((movie) => {
-        tableTemplate += ` 
+      result.forEach((movie) => {
+        tableTemplate += `
```

Comment:
Split this code to a template as function. So we can use it as below:

tableTemplate += MovieTemplate.renderTable(move);

Reply:
Updated!